### PR TITLE
Setup project base

### PR DESCRIPTION
**Pull request by ScottyJackson**
I built a little bit on the basic 'hello world' app we had. 

Namely this included restructuring the project by better defining the paths for the front and backend (backend code lived in `/application`, frontend in `/src`). 

I also introduced an entry point for the backend (`manage.py`), and started building out the home page, adding routes across the stack, and some basic components for the frontend.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 venv/
 __pycache__/
 node_modules/
-static/
+static/dist/
+package-lock.json
```
Comment (mbowen13) [inline .gitignore]: Do not ignore package-lock

File: package.json [hunk 2]
```diff
@@ -4,8 +4,9 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "start": "FLASK_APP=app.py flask run",
-    "build": "webpack",
+    "start": "python3 manage.py runserver",
```
Comment (mbowen13) [inline package.json:7]: @jkdf2 Can you take a look at the changes for start and build?

File: .gitignore (hunk 1 above)
Comment (mbowen13) [inline .gitignore]: Do not ignore package-lock

File: webpack.config.js [hunk 3]
```diff
@@ -1,10 +1,19 @@
-var webpack = require("webpack");
+const webpack = require('webpack')
```
Comment (mbowen13) [inline webpack.config.js]: @jkdf2 Can we make single quotes our convention and set a rule in the ESlint config?

**Review by mbowen13 — CHANGES_REQUESTED** [review]
Good work building out some of these components. 

There are a few things that need to be addressed before I can merge though. 

Please see my comments.

File: webpack.config.js (hunk 3 above)
Comment (taylorkline) [reply webpack.config.js]: @mbowen13 All style is handled through Prettier and yes it is possible to set our project to single quotes, but this is inadvisable especially in combination with JSX. Please see https://github.com/prettier/prettier/blob/master/Rationale.md

File: package.json [hunk 4]
```diff
@@ -4,8 +4,9 @@
...
+    "dev": "webpack -d --watch",
+    "build": "webpack --progress --colors",
     "test": "echo \"No tests added.\"",
     "eslint-check-prettier-config": "eslint --print-config .eslintrc.json | eslint-config-prettier-check",
     "check-style": "eslint --rule 'prettier/prettier: error' 'js/**'"
```
Comment (taylorkline) [inline package.json]: Please update check style to check `src/**` and ensure all tests are ran and pass on Travis.

**Comment by taylorkline** [comment]
I can't comment on `start` and `build`; the process was quite a bit different with React Native. I don't fully understand webpack nor Babel yet.

**Review by taylorkline — CHANGES_REQUESTED** [review]
Do you mind replacing PlaylistR with Playlistr? As in Flickr, Tumblr, etc.

**Comment by ScottyJackson** [comment]
Fixed all the issues above.

**Comment by taylorkline** [comment]
Nice! lgtm

**Review by mbowen13 — APPROVED** [review]

**Pull request merged**